src/cil.ml

```diff
@@ -2164,11 +2189,7 @@ let rec getInteger (e:exp) : cilint option =
       let mkInt ik n = Some (fst (truncateCilint ik n)) in
       match unrollType t, getInteger e with
       | TInt (ik, _), Some n -> mkInt ik n
-      | TPtr _, Some n -> begin
-	  match !upointType with
-	    TInt (ik, _) -> mkInt ik n
-	  | _ -> raise (Failure "pointer size unknown")
-        end
+      (* "integer constant expressions" may not cast to ptr *)
```

Comment:
What about the nullPtr?

Reply:
That's handled by `isNullPtrConstant` itself, which indirectly uses this.

According to the standard a `(void*)0` is not an _integer constant expression_ because it involves a cast to a non-integer type: https://en.cppreference.com/w/c/language/constant_expression#Integer_constant_expression.